### ISSUE TITLE
[MusicBERT] Conflicts in encoding OctupleMIDI tuples to OctupleMIDI string

### ISSUE THREAD

**Issue by tripathiarpan20** [microsoft/muzic#58]
Hi!

In order to encode a MIDI file to OctupleMIDI, I followed the script `musicbert/preprocess.py`. 

The OctupleMIDI tuples were obtained by running the following snipped as `e`:
![image](https://user-images.githubusercontent.com/42506819/177046887-e01dca92-6379-4c44-ac46-a6db50e06f94.png)

However, attempting to convert `e` to OctupleMIDI string can be done by two ways:
- Usage of  [Lines 370-398](https://github.com/microsoft/muzic/blob/f833b4f383e3df06235913fd50d94c2600e7af23/musicbert/preprocess.py#L370-L398) of `musicbert/preprocess.py` and considering `output_str_list[0]` as OctupleMIDI string.
- Usage of `encoding_to_str` method in `musicbert/preprocess.py` in [Line 429](https://github.com/microsoft/muzic/blob/main/musicbert/preprocess.py#L429)

But as seen in encircled text below there is some difference between these two in Measure field:
![image](https://user-images.githubusercontent.com/42506819/177047086-27ff3553-6a6a-4bd7-a455-437e1afb5d69.png)

Which approach was used during the pretraining of models? 
If I choose the wrong approach, the pretrained model would most likely give wrong embeddings.

Thanks in advance!

**Comment by mlzeng** [comment]
Since long sequences are randomly cropped, a random offset is added in the measure field during pre-train for normalization. Using encoding_to_str is OK in your case.
https://github.com/microsoft/muzic/blob/f833b4f383e3df06235913fd50d94c2600e7af23/musicbert/preprocess.py#L384

**Comment by tripathiarpan20** [comment]
I see, noted with thanks!
